backend/src/handlers_prelude/github_pull_request.rs

```diff
@@ -0,0 +1,54 @@
+//! Github Pull Request are sent here.
+use axum::{extract::State, http::StatusCode, Json, Router};
+use axum::routing::post;
+use tracing::{error, info};
+use serde_json::json;
+use crate::gh::create_pull_request;
+use crate::AppState;
+use color_eyre::Result;
+
+#[derive(serde::Deserialize, Debug)]
+pub struct CreatePRRequest {
+    pub head_branch: String,    // The branch containing changes
+    pub base_branch: String,     // The branch to merge into
+    pub title: String,
+    pub description: String,
+}
+
+pub async fn create_pull_request_handler(
+    State(state): State<AppState>,
+    Json(payload): Json<CreatePRRequest>,
+) -> Result<(StatusCode, Json<serde_json::Value>), (StatusCode, Json<serde_json::Value>)> {
```

Comment:
I would prefer if this type was more explicitly defined. Ideally it would be a full struct definition that implements `Serialize`, but if you feel that's not prodent, defining it as a `serde_json::Object` would also be acceptable.